### ISSUE TITLE
doctest failure with data-reify-0.6.2 (but not with data-reify-0.6.1)

### ISSUE THREAD

**Issue by RyanGlScott** [ekmett/ad#89]
On the `master` branch of `ad`, running the doctests against `data-reify-0.6.2` will produce the following failures:

```
Test suite doctests: RUNNING...
/home/rgscott/Documents/Hacking/Haskell/travis-maintenance/checkout/ekmett/ad/src/Numeric/AD/Newton/Double.hs:125: failure in expression `rosenbrock (conjugateGradientDescent rosenbrock [0, 0] !! 5) < 0.1'
expected: True
 but got: False
          ^

/home/rgscott/Documents/Hacking/Haskell/travis-maintenance/checkout/ekmett/ad/src/Numeric/AD/Newton.hs:291: failure in expression `rosenbrock (conjugateGradientDescent rosenbrock [0, 0] !! 5) < 0.1'
expected: True
 but got: False
          ^

Examples: 122  Tried: 122  Errors: 0  Failures: 2
```

What's more, these failures do not occur if built against `data-reify-0.6.1`. Here are the respective values of `rosenbrock (conjugateGradientDescent rosenbrock [0, 0] !! 5)` when built against each version of `data-reify`:

* `data-reify-0.6.1`:

  ```
  λ> rosenbrock (Numeric.AD.Newton.conjugateGradientDescent rosenbrock [0, 0] !! 5)
  3.124242390724632e-2
  ```
* `data-reify-0.6.2`

  ```
  λ> rosenbrock (Numeric.AD.Newton.conjugateGradientDescent rosenbrock [0, 0] !! 5)
  0.2093908903465224
  ```

(You'll get the same results if you use `Numeric.AD.Newton.Double.conjugateGradientDescent` instead.)

I've bisected this to ku-fpg/data-reify#8, but it's not yet clear to me why this causes the regression. Until I figure out why, I'll restrict the test suite to `data-reify < 0.6.2`.

**Comment by RyanGlScott** [comment]
It is also worth noting that this test suite failure does not occur with the latest Hackage release of `ad` (4.4). This failure only started happening in the `master` branch of `ad` after commit ed1e015f418b004348a210a9a1ded5d5b2415ec1.

**Comment by gdeest** [comment]
This was a long time ago, but no, I don't think this is expected — I'll have a look this weekend to see if I figure it out.

**Comment by RyanGlScott** [comment]
I've uploaded `data-reify-0.6.3` to Hackage, which restores the old (pre-0.6.2) behavior, fixing the `ad` test suite as a consequence. Moreover, I have deprecated `data-reify-0.6.2` on Hackage so that `cabal`'s constraint solver is disinclined to choose it in build plans. As a result, there should no longer be any need to explicitly exclude `data-reify-0.6.2` in `ad.cabal`, so I'm deeming this issue resolved.
